watch/watch.go

```diff
@@ -82,14 +82,18 @@ func (w *watchable) Watch() (interface{}, Watch, error) {
 	}
 
 	c := make(chan struct{}, 1)
-	if w.value != nil {
-		c <- struct{}{}
-	}
-
+	notify := w.value != nil
 	w.active = append(w.active, c)
-	closeFn := w.closeFunc(c)
 	w.Unlock()
 
+	if notify {
+		select {
```

Comment:
Yeah we need the select - It's already been added to all the slice of ch's, so technically it's possible that an update is registered to it after we unlock, but before we push to it here.